cdap-app-templates/cdap-etl/cdap-data-streams/src/main/java/co/cask/cdap/datastreams/DataStreamsPipelineSpecGenerator.java

```diff
@@ -53,8 +54,15 @@ public DataStreamsPipelineSpec generateSpec(DataStreamsConfig config) throws Inv
       .setStopGracefully(config.getStopGracefully())
       .setIsUnitTest(config.isUnitTest())
       .setCheckpointsDisabled(config.checkpointsDisabled());
-    if (config.getCheckpointDir() != null) {
-      specBuilder.setCheckpointDirectory(config.getCheckpointDir());
+    String checkpointDir = config.getCheckpointDir();
+    if (checkpointDir != null) {
```

Comment:
You only need to perform the check if checkpoint is not disabled, and if checkpoint is not disabled, this value must not be null, right?